Example/Auth/Sample/MainViewController+OAuth.m

```diff
@@ -307,6 +318,86 @@ - (void)signInYahooHeadfulLite {
   }];
 }
 
+- (ASAuthorizationAppleIDRequest *)appleIDRequestWithState:(NSString *)state API_AVAILABLE(ios(13.0)) {
+  ASAuthorizationAppleIDRequest *request = [[[ASAuthorizationAppleIDProvider alloc] init] createRequest];
+  request.requestedScopes = @[ASAuthorizationScopeEmail, ASAuthorizationScopeFullName];
+  // request.nonce = @"REPLACE_ME_WITH_YOUR_NONCE";
+  request.state = state;
+  return request;
+}
+
+- (void)signInWithApple {
+  if (@available(iOS 13, *)) {
+    ASAuthorizationAppleIDRequest* request = [self appleIDRequestWithState:@"signIn"];
+
+    ASAuthorizationController* controller = [[ASAuthorizationController alloc] initWithAuthorizationRequests:@[request]];
+    controller.delegate = self;
+    controller.presentationContextProvider = self;
+    [controller performRequests];
+  }
+}
+
+- (void)linkWithApple {
+  if (@available(iOS 13, *)) {
+    ASAuthorizationAppleIDRequest* request = [self appleIDRequestWithState:@"link"];
+
+    ASAuthorizationController* controller = [[ASAuthorizationController alloc] initWithAuthorizationRequests:@[request]];
+    controller.delegate = self;
+    controller.presentationContextProvider = self;
+    [controller performRequests];
+  }
+}
+
+- (void)reauthenticateWithApple {
+  if (@available(iOS 13, *)) {
+    ASAuthorizationAppleIDRequest* request = [self appleIDRequestWithState:@"reauth"];
+
+    ASAuthorizationController* controller = [[ASAuthorizationController alloc] initWithAuthorizationRequests:@[request]];
+    controller.delegate = self;
+    controller.presentationContextProvider = self;
+    [controller performRequests];
+  }
+}
+
+- (void)authorizationController:(ASAuthorizationController *)controller didCompleteWithAuthorization:(ASAuthorization *)authorization API_AVAILABLE(ios(13.0)) {
+  ASAuthorizationAppleIDCredential* appleIDCredential = authorization.credential;
+  NSString *idToken = [NSString stringWithUTF8String:[appleIDCredential.identityToken bytes]];
+  FIROAuthCredential *credential = [FIROAuthProvider credentialWithProviderID:@"apple.com"
+                                                                      IDToken:idToken
+                                    // rawNonce:@"REPLACE_ME_WITH_YOUR_RAW_NONCE"
```

Comment:
This looks like it will cause a build failure. I think you can avoid commenting out these placeholder nonce strings, they're pretty obviously placeholders.

Reply:
Yeah, good catch!

Reply:
To clarify, I meant you should leave the parameters in, but not commented out (unless that causes issues).

Will defer to your judgment here though, since I haven't actually built and ran the code.

Reply:
It's able to build while can't actually sign in. I think leaving the parameters in makes sense.